### ISSUE TITLE
Indentation check gives false positive: '@' has incorrect indentation level 4

### ISSUE THREAD

**Issue by rdiachenko** [checkstyle/checkstyle#8980]
Checkstyle 8.37, google-java-format 1.9
config (taken from google_checks.xml - https://github.com/checkstyle/checkstyle/blob/checkstyle-8.37/src/main/resources/google_checks.xml#L243):
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>
```
code:
```java
interface Test {
  @ApiResponses({
    @ApiResponse(
        code = 200,
        message = "Successfully found resource",
        reference = "https://www.hl7.org/fhir/bundle.html")
  })
  void foo();
}
```
Checkstyle gives indentation violation:
```bash
$ java -jar checkstyle-8.37-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/rdiachenko/work/workspace/checkstyle-playground/Test.java:3:5: '@' has incorrect indentation level 4, expected level should be 6. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
but there is an empty diff when applying google-java-format:
```bash
$ java -jar google-java-format-1.9-all-deps.jar Test.java > /tmp/FormattedTest.java && diff -Naru Test.java /tmp/FormattedTest.java
$
```
**Note:** empty diff when applying google-java-format means it's not expected to have code formatted in a different way, however Indentation check gives false positive and requires the code to be formatted differently.

**Comment by romani** [comment]
Line 3 looks like new block, so +2 indentation is expected.

**Comment by abebeos** [comment]
Is this a regression in 8.37?

> but there is an empty diff when applying google-java-format:

Note that this phrase is misleading.

You mean:

"but google-java-format validates the file" (the consequence of an empty diff)

**Comment by rdiachenko** [comment]
@abebeos 
> Note that this phrase is misleading.

Empty diff when applying google-java-format means it's not expected to have code formatted in a different way, however Indentation check gives false positive and requires the code to be formatted differently.

> Is this a regression in 8.37?

I didn't check it against previous versions, so I don't know.

**Comment by abebeos** [comment]
> 
> 
> @abebeos
> 
> > Note that this phrase is misleading.
> 
> Empty diff [...]

No need to explain.

"google-java-format validates the file""

> > Is this a regression in 8.37?
> 
> I didn't check it against previous versions, so I don't know.

You should, this is standard-procedure when filing issues.

Additionally, this should be catched by your CI test-cases.

**Comment by rdiachenko** [comment]
@abebeos 
> Additionally, this should be catched by your CI test-cases.

Pull request is welcome

**Comment by abebeos** [comment]
> > Additionally, this should be catched by your CI test-cases.
> 
> Pull request is welcome

Of course, but CI should be provided by the core-team anyways.

The procedure (for the core-team) would be:

- File Issue
- Add a test-case to CI
- Place bounty

Just suggesting here, because not everything is about bugs and codes.

Dev-Processes/procedures are important, too.

**Comment by Abhishek-kumar09** [comment]
@romani Its a duplicate to https://github.com/checkstyle/checkstyle/issues/5951. It would be fixed within that scope.

**Comment by rnveach** [comment]
Confirmed the violation goes away.

````
$ cat TestClass.java
interface Test {
  @ApiResponses({
    @ApiResponse(
        code = 200,
        message = "Successfully found resource",
        reference = "https://www.hl7.org/fhir/bundle.html")
  })
  void foo();
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>

$ java -jar checkstyle-nightly-2020-12-06-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````
